gradient/api_sdk/s3_uploader.py

```diff
@@ -315,13 +315,14 @@ def __init__(self, api_key, uploader=None, logger=None, ps_client_name=None):
             logger=self.logger,
             ps_client_name=ps_client_name,
         )
-        self.uploader = uploader or S3PutFileUploader(logger=self.logger, ps_client_name=ps_client_name)
+        self.uploader = S3PutFileUploader(logger=self.logger, ps_client_name=ps_client_name)
```

Comment:
Shouldn't the default parameter still be used? I think you just need to update where `DeploymentWorkspaceDirectoryUploader` is initialized instead. I think that's here https://github.com/Paperspace/gradient-cli/blob/c69250f3a7f7ddeb8458967fb4408dbf308aed9c/gradient/api_sdk/workspace.py#L96 I believe the call to `self.WORKSPACE_UPLOADER_CLS` needs to be `self.uploader_cls` instead because its overriding the passed in one.

Reply:
made the change and tried to create deployment but request becomes POST and fails with `403 The request signature we calculated does not match the signature you provided. Check your key and signing method.`